pkg/datastore/pubsub/mqtt/mqtt.go

```diff
@@ -0,0 +1,202 @@
+package mqtt
+
+import (
+	"encoding/json"
+	"strconv"
+
+	mqtt "github.com/eclipse/paho.mqtt.golang"
+
+	"gofr.dev/pkg"
+	"gofr.dev/pkg/datastore"
+	"gofr.dev/pkg/datastore/pubsub"
+	"gofr.dev/pkg/errors"
+	"gofr.dev/pkg/gofr/types"
+	"gofr.dev/pkg/log"
+)
+
+type MQTT struct {
+	client mqtt.Client
+	logger log.Logger
+	config *Config
+}
+
+type Config struct {
+	Protocol                string
+	Hostname                string
+	Port                    int
+	Username                string
+	Password                string
+	ClientID                string
+	Topic                   string
+	QoS                     byte
+	Order                   bool
+	ConnectionRetryDuration int
+}
+
+// New establishes connection to Kafka using the config provided in KafkaConfig
+func New(config *Config, logger log.Logger) (pubsub.PublisherSubscriber, error) {
+	pubsub.RegisterMetrics()
+
+	options := mqtt.NewClientOptions()
```

Comment:
there are some configs related to the client connection:
`ConnectRetry`, `AutoReconnect`.
Can you kindly check the criticality of these and see if we need these?

Reply:
We are doing manual retry connection like other databases so I think this might not be required